projectq/backends/_azure/_azure_quantum_test.py

```diff
@@ -0,0 +1,645 @@
+            use_hardware=False,
+            target_name='invalid-target',
+            workspace=workspace
+        )
+        assert False
+    except AzureQuantumTargetNotFoundError:
+        assert True
+
+
+@pytest.mark.parametrize(
+    "use_hardware, target_name, provider_id, current_availability",
+    [
+        (False, 'ionq.simulator', 'ionq', 'Available'),
+        (True, 'ionq.qpu', 'ionq', 'Available'),
+        (False, 'quantinuum.hqs-lt-s1-apival', 'quantinuum', 'Available'),
+        (False, 'quantinuum.hqs-lt-s1-sim', 'quantinuum', 'Available'),
+        (True, 'quantinuum.hqs-lt-s1', 'quantinuum', 'Degraded'),
+    ],
+)
+def test_current_availability(use_hardware, target_name, provider_id, current_availability):
+    workspace = mock_workspace()
+
+    backend = AzureQuantumBackend(
+        use_hardware=use_hardware,
+        target_name=target_name,
+        workspace=workspace
+    )
+
+    assert backend.current_availability == current_availability
+
+
+@pytest.mark.parametrize(
+    "use_hardware, target_name, provider_id, average_queue_time",
+    [
+        (False, 'ionq.simulator', 'ionq', 1000),
+        (True, 'ionq.qpu', 'ionq', 2000),
+        (False, 'quantinuum.hqs-lt-s1-apival', 'quantinuum', 3000),
+        (False, 'quantinuum.hqs-lt-s1-sim', 'quantinuum', 4000),
+        (True, 'quantinuum.hqs-lt-s1', 'quantinuum', 5000),
+    ],
+)
+def test_average_queue_time(use_hardware, target_name, provider_id, average_queue_time):
+    workspace = mock_workspace()
+
+    backend = AzureQuantumBackend(
+        use_hardware=use_hardware,
+        target_name=target_name,
+        workspace=workspace
+    )
+
+    assert backend.average_queue_time == average_queue_time
+
+
+@pytest.mark.parametrize(
+    "use_hardware, target_name, provider_id",
+    [
+        (False, 'ionq.simulator', 'ionq'),
+        (True, 'ionq.qpu', 'ionq')
+    ],
+)
+def test_run_ionq_get_probabilities(use_hardware, target_name, provider_id):
+    projectq.backends._azure._azure_quantum.send = mock.MagicMock(
+        return_value={
+            'histogram': {
+                '0': 0.5,
+                '1': 0.0,
+                '2': 0.0,
+                '3': 0.0,
+                '4': 0.0,
+                '5': 0.0,
+                '6': 0.0,
+                '7': 0.5
+            }
+        }
+    )
+
+    workspace = mock_workspace()
+
+    backend = AzureQuantumBackend(
+        use_hardware=use_hardware,
+        target_name=target_name,
+        workspace=workspace,
+        verbose=True
+    )
+
+    mapper = BasicMapperEngine()
+    max_qubits = 3
+
+    mapping = {}
+    for i in range(max_qubits):
+        mapping[i] = i
+
+    mapper.current_mapping = mapping
+
+    main_engine = MainEngine(
+        backend=backend,
+        engine_list=[mapper],
+        verbose=True
+    )
+
+    circuit = main_engine.allocate_qureg(3)
+    q0, q1, q2 = circuit
+
+    H | q0  # noqa
+    CX | (q0, q1)  # noqa
+    CX | (q1, q2)  # noqa
```

Comment:
These `noqa` are not be required since we have a general exclusion rule for `useless-statements`

```suggestion
    H | q0
    CX | (q0, q1)
    CX | (q1, q2)
```